eclass/dotnet-utils.eclass

```diff
@@ -0,0 +1,216 @@
+# @INTERNAL
+# @DESCRIPTION:
+# Sets to the valid path of a dotnet executable
+
+# @ECLASS_VARIABLE: DOTNET_OUTPUT
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Path of the output directory, if unset, is set during src_compile to:
+# "${WORKDIR}"/${P}_net${DOTNET_SLOT}_${DOTNET_CONFIGURATION}/${PN}
+
+# @ECLASS_VARIABLE: DOTNET_CONFIGURATION
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Configuration value passed to dotnet in the compile phase.
+# Defaults to Debug or Release, depending on the debug useflag.
+
+# @ECLASS_VARIABLE: DOTNET_CLI_TELEMETRY_OPTOUT
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Disables telemetry on dotnet.
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+
+# @ECLASS_VARIABLE: MSBUILDDISABLENODEREUSE
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Stops the dotnet node after the build is done.
+export MSBUILDDISABLENODEREUSE=1
+
+# @ECLASS_VARIABLE: DOTNET_NOLOGO
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Disables the header logo when running dotnet commands.
+export DOTNET_NOLOGO=1
+
+# Needed otherwise the binaries may break
+RESTRICT+=" strip"
+
+# @ECLASS_VARIABLE: NUGETS
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# String containing all nuget packages that need to be downloaded.
+# Used by nuget_uris()
+#
+# Example:
+# @CODE
+# NUGETS="
+# 	ImGui.NET-1.87.2
+# 	Config.Net-4.19.0
+# "
+#
+# inherit dotnet-utils
+#
+# ...
+#
+# SRC_URI="$(nuget_uris)"
+# @CODE
+
+# @FUNCTION: nuget_uris
+# @USAGE: <nuget...>
+# @DESCRIPTION:
+# Generates the URIs to put in SRC_URI to help fetch dependencies.
+# If no arguments provided, uses NUGETS variable.
+nuget_uris() {
+	local -r regex='^([a-zA-Z0-9_.-]+)-([0-9]+\.[0-9]+\.[0-9]+.*)$'
+	local nuget nugets
+
+	if (( $# != 0 )); then
+		nugets="${@}"
+	elif [[ ${NUGETS} ]]; then
+		nugets="${NUGETS}"
+	else
+		eerror "NUGETS variable is not defined and nothing passed as argument"
+		die "${FUNCNAME}: Can't generate SRC_URI from empty input"
+	fi
+
+	for nuget in ${nugets}; do
+		local name version url
+		[[ ${nuget} =~ ${regex} ]] || die "Could not parse name and version from nuget: $nuget"
+		name="${BASH_REMATCH[1]}"
+		version="${BASH_REMATCH[2]}"
+		url="https://api.nuget.org/v3-flatcontainer/${name}/${version}/${name}.${version}.nupkg"
+		echo "${url}"
+	done
+}
+
+# @FUNCTION: dotnet-utils_pkg_setup
+# @DESCRIPTION:
+# Sets up DOTNET_EXECUTABLE variables for later use in edotnet.
+dotnet-utils_pkg_setup() {
+	if use amd64; then
+		DOTNET_RUNTIME="linux-x64"
+	elif use arm; then
+		DOTNET_RUNTIME="linux-arm"
+	elif use arm64; then
+		DOTNET_RUNTIME="linux-arm64"
+	else
+		die "Unsupported arch: ${ARCH}"
+	fi
+
+	local _dotnet
+	for _dotnet in dotnet{,-bin}-${DOTNET_SLOT}; do
+		if type ${_dotnet} 1> /dev/null 2>&1; then
+			DOTNET_EXECUTABLE=${_dotnet}
+			break
+		fi
+	done
+
+}
+
+# @FUNCTION: edotnet
+# @USAGE: <command> [args...]
+# @DESCRIPTION:
+# Call dotnet, passing the supplied arguments.
+edotnet() {
+	debug-print-function ${FUNCNAME} "${@}"
+
+	local dotnet_args=(
+		-maxcpucount:$(makeopts_jobs)
```

Comment:
maxcpucount might be causing some bugs, see: https://bugs.gentoo.org/890297#c8

Reply:
What's ideal to do here, just drop it for now?

Reply:
> What's ideal to do here, just drop it for now?

I think we could hardcode it to 1 now and try to lift this restriction in the future.
Best solution is to ask some real .NET dev if bug #890297 is really caused by parallel build jobs...